### ISSUE TITLE
Footswitch1-F1.8 not recognised under WSL ( windows subsystem for linux )

### ISSUE THREAD

**Issue by bradphelan** [rgerganov/footswitch#42]
I have a PCSensor foot pedal. When plugged it windows 10 says it is a

Footswitch1-F1.8 

I've tried the software under WSL but when I do

footswitch -r 

I get

> 
> Cannot find footswitch with one of the supported VID:PID.
> Check that the device is connected and that you have the correct permissions to access it. [init(), footswitch.c:109]

Is this a WSL problem or the footswitch model is not supported?

**Comment by rgerganov** [comment]
Does WSL have the `lsusb` command? If so, can you paste the output when the switch is connected?

**Comment by bradphelan** [comment]
First I found the HID data under the windows driver list

![image](https://user-images.githubusercontent.com/17650/43819951-11f7b572-9ae5-11e8-9bcb-c817bcd9a9b3.png)
 
so 
VID = 0C45
PID = 7404

which seems to be on the list of supported ids

![image](https://user-images.githubusercontent.com/17650/43820003-31b4c896-9ae5-11e8-9e91-3fbda7d3abfc.png)

however running

sudo lsusb

returns no result. Maybe they are hidden from WSL ??

**Comment by rgerganov** [comment]
Yes, this is supported VID:PID but unfortunately it seems to be hidden in WSL. Please let me know if you manage to find a workaround.

**Comment by bradphelan** [comment]
I think there is no direct USB support under WSL only mounting of USB drives.

**Comment by bradphelan** [comment]
I found this but I'm not sure it's relevant. Is the footswitch a "serial" device as described in the link

https://blogs.msdn.microsoft.com/wsl/2017/04/14/serial-support-on-the-windows-subsystem-for-linux/

**Comment by rgerganov** [comment]
No, it is a USB HID device.

**Comment by doadin** [comment]
I know this is old but its still open and has no solution, I recently tried to use the pcsensor software and it wouldnt work in windows. noticed this project and im trying to get it working in windows but no luck. 

`error writing data ( [usb_write(), E:\footswitch-1.0\footswitch.c:147]`

I have hidapi working though as I can get manufacture/product/serial number/indexed string(s) but usb_write doesnt work. is it possible usb_write(query) is wrong? or is this more possibly a permissions issue? I tried runing from an admin powershell. same thing happens. It would seem hidapi is able to read just not write.
